packages/beacon-node/src/sync/sync.ts

```diff
@@ -164,7 +164,7 @@ export class BeaconSync implements IBeaconSync {
       case RangeSyncStatus.Idle:
         return SyncState.Stalled;
       default:
-        return SyncState.Stalled;
+        throw new Error("Unreachable code");
```

Comment:
could also throw an error which includes the value, just in case this actually happens makes debugging easier

Reply:
The funny thing is Typescript complains `rangeSyncState.status` to be `never` type so can't be used.

Reply:
makes sense from typescript pov this can never happen, let's keep it as is